src/commands/functions-config-export.ts

```diff
@@ -0,0 +1,383 @@
+function writeEmptyEnvs(basePath: string, header: string, projects: Project[]): string[] {
+  return writeEnvs(
+    basePath,
+    header,
+    projects,
+    projects.map(() => [])
+  );
+}
+
+async function copyFilesToDir(srcFiles: string[], destDir: string): Promise<string[]> {
+  const destFiles = [];
+  for (const file of srcFiles) {
+    const targetFile = path.join(destDir, path.basename(file));
+    if (fs.existsSync(targetFile)) {
+      const overwrite = await promptOnce(
+        {
+          type: "confirm",
+          name: "overwrite",
+          default: true,
+          message: `${targetFile} already exists. Overwrite file?`,
+        },
+        { nonInteractive: false }
+      );
+      if (!overwrite) {
+        logBullet(`Skipping ${targetFile}`);
+        continue;
+      }
+    }
+    fs.copyFileSync(file, targetFile);
+    destFiles.push(targetFile);
+  }
+  return destFiles.filter((f) => f.length > 0);
+}
+
+export default new Command("functions:config:export")
+  .description("Export environment config as environment variables in dotenv format")
+  .before(requirePermissions, [
+    "runtimeconfig.configs.list",
+    "runtimeconfig.configs.get",
+    "runtimeconfig.variables.list",
+    "runtimeconfig.variables.get",
+  ])
+  .action(async (options: any) => {
+    const allProjects = getAllProjects(options);
+
+    if (allProjects.length == 0) {
+      throw new FirebaseError(
+        "Didn't find any project in the current directory. " +
+          "Are you in a firebase project directory?"
+      );
+    }
+
+    const projects: Project[] = [];
+    for (const project of allProjects) {
+      if ((await checkRequiredPermission(project)) && (await checkReservedAlias(project))) {
+        projects.push(project);
+      }
+    }
+
+    logBullet(
+      "Importing functions configs from projects [" +
+        projects.map(({ projectId }) => `${clc.bold(projectId)}`).join(", ") +
+        "]"
+    );
+
+    const configs = await Promise.all(
+      projects.map(async ({ projectId }) => {
+        return await functionsConfig.materializeAll(projectId);
+      })
+    );
+    logger.debug("Loaded function configs: " + JSON.stringify(configs));
+
+    let prefix = "";
+    let envs = [];
+    while (true) {
+      const result = configsToEnvs(projects, configs, prefix);
+      envs = result.envs;
+
+      if (result.errMsg.length == 0) {
+        break;
+      }
+      prefix = await promptForPrefix(result.errMsg);
+
+      envs = [];
+    }
+
+    const header = `# Exported firebase functions:config:export command on ${new Date().toLocaleDateString()}`;
+    const tmpdir = fs.mkdtempSync(os.tmpdir() + "dotenvs");
+    const tmpFiles = writeEnvs(tmpdir, header, projects, envs);
+    tmpFiles.push(...writeEmptyEnvs(tmpdir, header, [{ projectId: "" }, { projectId: "local" }]));
+    logger.debug(`Wrote tmp .env files: [${tmpFiles.join(",")}]`);
+
+    const functionsDir: string = options.config.get("functions.source", ".");
```

Comment:
Make sure this command works if you run it in a project directory or any subdir thereof. You shouldn't have to run this in `functions` to make it work. See e.g. [resolveProjectPath](https://github.com/firebase/firebase-tools/blob/eedf0c469aec6a78d4965df859be08d5e79adbbc/src/projectPath.ts#L11).

Reply:
Great point! Updated.